src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -624,6 +544,108 @@ def inner_Bethe_stopping(v, n_e):
         self._stopping_method = method
         self._stopping_power_interpolators = stopping_power_interpolators
 
+        self._log(f"Stopping module activated using the {method} method")
+
+    def _validate_grids(self) -> None:
+        """Validate input grids."""
+        for i, grid in enumerate(self.grids):
+            for q in grid.quantities:
+                # Check that there are no infinite values
+                if not np.isfinite(grid[q].value).all():
+                    raise ValueError(
+                        f"Input arrays must be finite: {q} contains "
+                        "either NaN or infinite values."
+                    )
+
+                # Check that the max values on the edges of the arrays are
+                # small relative to the maximum values on that grid
+                #
+                # Array must be dimensionless to re-assemble it into an array
+                # of max values like this
+                arr = np.abs(grid[q]).value
+                edge_max = np.max(
+                    np.array(
+                        [
+                            np.max(a)
+                            for a in (
+                                arr[0, :, :],
+                                arr[-1, :, :],
+                                arr[:, 0, :],
+                                arr[:, -1, :],
+                                arr[:, :, 0],
+                                arr[:, :, -1],
+                            )
+                        ]
+                    )
+                )
+
+                if edge_max > 1e-3 * np.max(arr):
+                    unit = grid.recognized_quantities()[q].unit
+                    warnings.warn(
+                        "Quantities should go to zero at edges of grid to avoid "
+                        f"non-physical effects, but a value of {edge_max:.2E} {unit} was "
+                        f"found on the edge of the {q} array of grid {i}. Consider applying a "
+                        "envelope function to force the quantities at the edge to go to "
+                        "zero.",
+                        RuntimeWarning,
+                    )
+
+    def _setup_for_interpolator(self) -> None:
+        """Prepare grids for interpolation.
+
+        This will be called at the beginning of run, since the required
+        quantities depends on how the Tracker has been set up.
+        """
+        # Assemble an array of the resolutions of all of the grids
+        self._grid_resolutions = np.array(
+            [grid.grid_resolution.to(u.m).value for grid in self.grids]
+        )
+
+        # Determine which quantities should be interpolated from each grid
```

Comment:
This code goes through and figures out which quantities to interpolate on which grids, then creates a dictionary with arrays to hold those values. Reserving that memory now so we don't have to keep creating variables later.